persistence/persistence-api/src/main/java/org/cbioportal/persistence/TreatmentRepository.java

```diff
@@ -27,4 +27,7 @@
 
     @Cacheable(cacheResolver = "generalRepositoryCacheResolver", condition = "@cacheEnabledConfig.getEnabled()")
     public Integer getSampleCount(List<String> studies);
+
+    @Cacheable(cacheResolver = "generalRepositoryCacheResolver", condition = "@cacheEnabledConfig.getEnabled()")
+    public Set<String> getFilteredStudyIds(List<String> studyIds, ClinicalEventKeyCode key);
```

Comment:
can we change this to cache at single study level?  i know this requires then querying at single study level but it's a tradeoff.

Reply:
Sure. Fixed